modules/reindex/build.gradle

```diff
@@ -26,6 +26,7 @@ integTest {
   cluster {
     // Whitelist reindexing from the local node so we can test it.
     setting 'reindex.remote.whitelist', 'myself'
+    plugin 'optional-auth', project(':test:plugins:optional-auth')
```

Comment:
indeed, this is going to work for es core only. They are not going to have the special test plugin. I would talk to them and discuss how to move forward. Maybe the auth tests need to be separated out and rely on a new test feature called "basic-auth". The runners that implement this feature rely on basic auth based on given user and password, regardless of how that is implemented (through plugin, or through proxy). How does that sound?